src/StackExchange.Redis.Extensions.Core/StackExchangeRedisCacheClient.cs

```diff
@@ -628,8 +628,13 @@ public IEnumerable<string> SearchKeys(string pattern)
 
 			foreach (var endpoint in endPoints)
 			{
-				var dbKeys = Database.Multiplexer.GetServer(endpoint).Keys(Database.Database, pattern);
+				var server = Database.Multiplexer.GetServer(endpoint);
+				if (!server.IsConnected || !server.Features.Scan)

```

Comment:
As a rule of thumb @EugeneKrapivin comment is correct. This is not a general case though. There are valid scenarios with Redis where certain instances get commands removed through configuration. In this case we might be interested in getting any key on any server willing to provide them.
Furthermore, this same check is happening internally, except that because this method is traversing all hosts in the Redis setup, any hiccup in one of them will render the functionality unusable for any other server.
In our environment we have a server registered that doesn't accept connections, a perfectly valid production scenario. All other servers should be able to handle out their keys collection to serve SearchKeys.